### PR TITLE
[APO-1265] Add comprehensive test suite for WebSearchNode

### PR DESCRIPTION
**Pull request by VJagiasi**
Adds comprehensive test suite for WebSearchNode with requests_mock integration. Tests cover success scenarios, error handling, input validation, and edge cases

## Test Coverage 
  **Core Functionality:**
  - ✅ Successful search with SerpAPI response parsing
  - ✅ Text output formatting (title/snippet concatenation)
  - ✅ URL extraction from search results  
  - ✅ Raw results preservation

  **Error Handling:**
  - ✅ 401 authentication errors → INVALID_INPUTS
  - ✅ 429 rate limiting → PROVIDER_ERROR
  - ✅ 500+ server errors → PROVIDER_ERROR
  - ✅ Invalid JSON responses → PROVIDER_ERROR
  - ✅ SerpAPI error responses → PROVIDER_ERROR

  **Input Validation:**
  - ✅ Empty query validation
  - ✅ Missing API key validation
  - ✅ Invalid num_results validation

  **Edge Cases:**
  - ✅ Empty organic_results handling
  - ✅ Missing title/snippet/link fields
  - ✅ Network timeout handling
  - ✅ Location parameter support
  - ✅ User-Agent header verification

### CODE REVIEW

File: src/vellum/workflows/nodes/displayable/web_search_node/tests/test_node.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+from vellum.workflows.state.base import StateMeta
+
+from ..node import WebSearchNode
+
+
+@pytest.fixture
+def mock_context():
+    """Create a mock context with vellum_client."""
+    context = MagicMock()
+    context.vellum_client._client_wrapper.get_headers.return_value = {"User-Agent": "vellum-python-sdk/1.0.0"}
```
Comment (dvargasfuertes) [inline src/vellum/workflows/nodes/displayable/web_search_node/tests/test_node.py]: [nit] we have a `vellum_client` fixture that we're using that we could add this support to and use instead

File: src/vellum/workflows/nodes/displayable/web_search_node/tests/test_node.py [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+        pass
+
+    class TestableWebSearchNode(WebSearchNode):
+        query = Inputs.query
+        api_key = Inputs.api_key
+        num_results = Inputs.num_results
+
+    state = State(meta=StateMeta(workflow_inputs=Inputs(query="test query", api_key="test_api_key", num_results=3)))
+    node = TestableWebSearchNode(state=state)
+    node._context = mock_context
```
Comment (dvargasfuertes) [inline src/vellum/workflows/nodes/displayable/web_search_node/tests/test_node.py]: [nit] let's pass in the context using `node = TestableWebSearchNode(state=state, context=mock_context)`

**Review by dvargas92495 — APPROVED** [review]
looks great!
